### ISSUE TITLE
Executable.getAnnotatedParameterTypes throws exception when trying to get annotated parameter type in an inner class

### ISSUE THREAD

**Issue by felipepontes-research** [eclipse/openj9#1852]
An exception is thrown when executing the following program:

```
public class A {
    class Inner {
        Inner (Iterator<String> delegate) {
        }
    }
    public static void main(String[] args) {
        Class<?>[] classes = A.class.getDeclaredClasses();
	System.out.println(classes[0].getDeclaredConstructors()[0].getParameters()[0].getDeclaringExecutable().getAnnotatedParameterTypes()[0].getType());
    }
}
```

**affected versions:**

openjdk version "1.8.0_162"
OpenJDK Runtime Environment (build 1.8.0_162-b12)
Eclipse OpenJ9 VM (build openj9-0.8.0, JRE 1.8.0 Linux amd64-64 Compressed References 20180315_120 (JIT enabled, AOT enabled)
OpenJ9 - e24e8aa9
OMR - 3e8296b4
JCL - ee1e77df1d based on jdk8u162-b12)

**steps to reproduce:**

	1. compile above program
	2. run it using openj9 java

**current result:**

Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: Array index out of range: 1
	at java.lang.reflect.Executable.getAllGenericParameterTypes(Executable.java:318)
	at java.lang.reflect.Executable.getAnnotatedParameterTypes(Executable.java:693)
	at A.main(A.java:8)

**expected result:**

class A

**Comment by JasonFengJ9** [comment]
Not able to reproduce the exception reported.

@felipepontes-research could you provide the actual `java` file and command line options? thanks.

**Comment by felipepontes-research** [comment]
I think it was because I don't formatted sample code properly. So, ```Inner (Iterator<String> delegate)``` was converted into ```Inner (Iterator delegate)```. I updated it.

**Comment by JasonFengJ9** [comment]
Thanks, the exception reproduced.

**Comment by aguibert** [comment]
+1 for this issue, we ran into this on @OpenLiberty when attempting to upgrade from Hibernate Validator 6.0.9 to 6.0.10

**Comment by JasonFengJ9** [comment]
Note: working on a fix.

**Comment by JasonFengJ9** [comment]
This should be closed by #2053 (the clause closing multiple issues wasn't in proper format).
